clang/lib/Driver/ToolChains/Gnu.cpp

```diff
@@ -571,6 +564,15 @@ void tools::gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   bool NeedsXRayDeps = addXRayRuntime(ToolChain, Args, CmdArgs);
   addLinkerCompressDebugSectionsOption(ToolChain, Args, CmdArgs);
   AddLinkerInputs(ToolChain, Inputs, Args, CmdArgs, JA);
+
+  if (D.IsFlangMode()) {
+    CmdArgs.push_back("-lFortran_main");
+    CmdArgs.push_back("-lFortranRuntime");
+    CmdArgs.push_back("-lFortranDecimal");
+    CmdArgs.push_back("-lFortranRuntime");
```

Comment:
Do you need to duplicate the `-lFortranRuntime` here ?

Reply:
No, that's a leftover from debugging. Good catch, thanks! I'll update this in a few mins.